### ISSUE TITLE
Can a simple tutorial be provided for non-professionals

### ISSUE THREAD

**Issue by JadeVane** [ncarlier/readflow#48]
Thanks for make this amazing project, I have been searching a self-hosted read it later project which was made by golang for a long time.

I found readflow yesterday and tried to deploy it to my server for several hours, but still not working, I'm not a professional programmer, I don't know much about the instructions in the configuration file, I tried my best to understand what the config file was saying and google what I didn't understand, but no luck.

In fact, it took me almost 2 hours just to google how to successfully compile binaries when `go get` fails. Maybe a layman like me is not the target user of this project, but I still hope to try this project, I will try to explain the problems I have encountered and hope it can be solved, thank you again for developing this project.

config.toml

```bash
[global]
db = "postgres://username:password@localhost/databasename?sslmode=disable"
listen_addr = ":9021"
metrics_listen_addr = ""
authn = "mock"
public_url = ""
secret_salt = "${READFLOW_SECRET_SALT}"
block_list = "${READFLOW_BLOCK_LIST}"
[integration]
external_event_broker_uri = "${READFLOW_BROKER_URI}"
external_web_scraper_url = "${READFLOW_WEB_SCRAPER_URL}"
image_proxy_url = "${READFLOW_IMAGE_PROXY_URL}"
avatar_provider = "${READFLOW_AVATAR_PROVIDER}"
[integration.sentry]
dsn_url = "${READFLOW_SENTRY_DSN}"
[integration.pocket]
consumer_key = "${READFLOW_POCKET_CONSUMER_KEY}"
```

run it

```bash
readflow --config config.toml
```

the output

```bash
{"level":"info","component":"database","uri":"postgres://readflow:xxxxx@localhost/readflow?sslmode=disable","time":1658973321,"message":"using PostgreSQL database"}
{"level":"info","component":"cache","uri":"boltdb:///tmp/readflow.cache","time":1658973321,"message":"using BoltDB cache"}
{"level":"info","component":"service","rev":1,"time":1658973321,"message":"properties loaded"}
{"level":"info","time":1658973321,"message":"using Mock as authentication backend"}
{"level":"info","component":"api","provider":"https://robohash.org/{seed}?set=set4&size=48x48","time":1658973321,"message":"using Avatar provider"}
{"level":"info","listen":":9021","time":1658973321,"message":"server started"}
```

I can't get the web page in the browser when visit `http://ip:9021`, I can only see a json file

```bash
{"version":"snapshot","authority":"mock","vapid":"BASakRma0VQ2znpuOtnaSVsLg34PAatW6dUMLi1t55vdU2Awxnulnb5UmPR4yEiOq5xamXz6YVVoR8QNhq8rDQc"}
```

**Comment by ncarlier** [comment]
Hello, thank you for your interest in this project. 

Actually, your installation is almost complete. You still need two things: 

- Use another authentication method than "mock". You can use a proxy and delegate the authentication to a reverse proxy such as Apache, NGINX or Traefik. It is almost the same as for this Grafana tutorial: https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/auth-proxy/
A better option is to use a OpenID Connect provider.

- finally you need the user interface. It is separate from the Golang binary. You need to start the UI. For this you can build the UI with `npm run build` then expose the `./build` directory with a static web server such as Apache or NGINX or you can use directly NodeJS with `npm start`. You must set some environment variables to match the API URL and authentication provider: https://github.com/ncarlier/readflow/tree/master/ui 

FYI, you can also use Docker Compose (`docker-compse up`) to set up a fast instance.

**Comment by JadeVane** [comment]
Thanks for your reply, I followed the steps you said and it still doesn't work but has made great progress.

- **Use another authentication method**
I set `authn = "proxy"` in config.toml, and copy the configuration file of nginx from this repo, that's all what i has done. This should be the reason why it still can't be used, because I can't detect any connection between grafana's configuration file and readflow, in my opinion, the two are not much related

- **User interface**
I build it from the ui folder and it works fine, so i see the error message from web: `Unable to get user information: NetworkError when attempting to fetch resource.`

- **Docker**
I'd prefer to have a deeper understanding of what I'm doing, so at first I'd like to be able to run from source

readflow.conf

```bash
server {
    listen                443 ssl;
    server_name           readflow.domain.com;
    ssl_certificate       /etc/letsencrypt/live/readflow.domain.com/fullchain.pem;
    ssl_certificate_key   /etc/letsencrypt/live/readflow.domain.com/privkey.pem;
  
    gzip                 on;
    gzip_static          on; # allows pre-serving of .gz file if it exists 
    gzip_disable         "msie6"; # Disable for user-agent Internet explorer 6. Not supported.
    gzip_proxied         any; # enable gzip for all proxied requests
    gzip_buffers         16 8k; # number and size of buffers to compress a response
    gzip_http_version    1.1;
    gzip_min_length      256; # Only gzip files of size in bytes
    gzip_types           text/plain text/css application/javascript application/json x-javascript text/xml application/xml application/xml+rss text/javascript application/ vnd.ms-fontobject application/x-font-ttf font/opentype image/svg+xml image/x-icon;
    gunzip               on;
  
    location / {
      root                   /var/www/readflow/web;
      index                  index.html index.htm;
      try_files              $uri $uri/ /index.html;
      auth_basic             "Restricted";
      auth_basic_user_file    /etc/nginx/.htpasswd;
      proxy_set_header        X-WEBAUTH-USER $remote_user;
      proxy_set_header        Authorization "";
    }
  
    error_page               500 502 503 504  /50x.html;
    location = /50x.html {
      root                   /var/www/readflow/web;
    }
}

server
{
    listen                80;
    server_name           readflow.domain.com;
    return                301 https://readflow.domain.com$request_uri;
}
```

api.conf

```bash
server {
    listen                443 ssl;
    server_name           api.domain.com;
    ssl_certificate       /etc/letsencrypt/live/api.domain.com/fullchain.pem;
    ssl_certificate_key   /etc/letsencrypt/live/api.domain.com/privkey.pem;

    location /readflow {
        proxy_pass http://localhost:9021;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "Upgrade";
        proxy_set_header Host $host;

        access_log            /var/log/readflow-access.log main;
        error_log             /var/log/readflow-error.log;
    }
}
```

and the step to build ui

```bash
export REACT_APP_API_ROOT=https://api.domain.com/readflow
export REACT_APP_AUTHORITY=mock
export REACT_APP_CLIENT_ID=
export REACT_APP_REDIRECT_URL=https://read.domain.com/login
npm run build
```

**Comment by ncarlier** [comment]
The authentication filter should be on the API and not on the UI.
I imagine that the domain names are examples.
For testing purpose I suggest you to use localhost and different ports. Just in order to validate the setup.

But let's imagine that your domain is `readflow.domain.com`:
UI domain should be `readflow.domain.com` and API domain should be `api.readflow.domain.com`.
BTW your REACT_APP_REDIRECT_URL should be `https://readflow.domain.com/login`

**Comment by JadeVane** [comment]
You are right, deploying the project on localhost for testing is a good idea, finally I found the key. The `public_url` must be configured correctly. 

But there is a new issue. I set `public_url = "http://localhost:9021"` locally and it works fine, however, the api is proxied by nginx with domain `https://api.domain.com/readflow` on the server, and it doesn't work, so what is the key for the `public_url`?
